### PR TITLE
Formatter expand/collapse/ignore empty XML tags

### PR DESCRIPTION
**Pull request by angelozerr**
Formatter expand/collapse/ignore empty XML tags

See https://github.com/redhat-developer/vscode-xml/issues/219

Signed-off-by: azerr <azerr@redhat.com>

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -106,6 +106,17 @@
           "default": true,
           "description": "Should the server log to client output"
         },
+        "xml.format.formatEmptyElements": {
```
Comment (fbricon) [inline package.json:109]: xml.format.emptyElements

there's already "format" in the key

File: README.md [hunk 2]
```diff
@@ -53,6 +53,7 @@ The following settings are supported:
 * `xml.catalogs` : Register XML catalog files.
 * `xml.logs.client` : Enable/disable logging to the Output view.
 * `xml.fileAssociations` : Associate XML Schemas to XML file patterns.
+* `xml.format.formatEmptyElements` : Expand/Collapse empty elements during formatting. Defaults to `none`.
```
Comment (fbricon) [inline README.md:56]: `xml.format.emptyElements`

File: package.json (hunk 1 above)
Comment (angelozerr) [reply package.json]: fixed in this PR and fixed on LemMinx side

File: README.md (hunk 2 above)
Comment (angelozerr) [reply README.md]: fixed

**Comment by xorye** [comment]
I found a minor edge case when I set xml.format.emptyElements to collapse:

range formatting

After range formatting, I think the result should be:
```
<a>
    <b>
</b>
</a>
```
as only the starting tag was to be formatted

**Comment by fbricon** [comment]
@angelozerr please see if this is fixable. maybe don't do the collapsing/expansion if the  selection is not self-contained, i.e it's malformed on its own.

**Comment by angelozerr** [comment]
I close this PR for https://github.com/redhat-developer/vscode-xml/pull/242 (I had some trouble with Eclipse Egit NPE bug)

**Comment by angelozerr** [comment]
> I found a minor edge case when I set xml.format.emptyElements to collapse:

Good catch!

> @angelozerr please see if this is fixable. maybe don't do the collapsing/expansion if the selection is not self-contained, i.e it's malformed on its own.

I fixed this issue, if element is not closed (when range doesn't cover the whole element, the collapse/expand is ignored for this element

**Pull request closed without merging**